### PR TITLE
Feature/submission on behalf

### PR DESCRIPTION
**Pull request by FelixRottler**
- draft PR for keeping track of the progress for submission on behalf

### CODE REVIEW

**Review by AliceBilc — APPROVED** [review]

File: services/submission/src/main/java/app/coronawarn/server/services/submission/checkins/EventCheckInProtectedReportsValidator.java [hunk 1]
```diff
@@ -27,7 +34,26 @@ public boolean verify(SubmissionPayload submissionPayload, ConstraintValidatorCo
...
+   * @param context           the validator context for keeping track of the constrain violations.
+   * @return whether the protected checkins of this submission payload contain all the same location id.
+   */
+  public boolean verifyHaveSameLocationIdHash(SubmissionPayload submissionPayload, ConstraintValidatorContext context) {
+    final boolean checkInsHaveSameLocationId =
+        submissionPayload.getCheckInProtectedReportsList().stream().map(CheckInProtectedReport::getLocationIdHash)
+            .distinct().count() == 1;
+    if (!checkInsHaveSameLocationId) {
+      logger.warn(SECURITY, "Checkins must contain items that all share the same location id");
+      addViolation(context, "Checkins protected reports must contain items that all share the same location id hash.s");
```
Comment (FelixRottler) [inline services/submission/src/main/java/app/coronawarn/server/services/submission/checkins/EventCheckInProtectedReportsValidator.java:54]:
```suggestion
      addViolation(context, "Checkins protected reports must contain items that all share the same location id hash.");
```

File: services/submission/src/main/java/app/coronawarn/server/services/submission/checkins/EventCheckInProtectedReportsValidator.java [hunk 2]
```diff
@@ -27,7 +34,26 @@ public boolean verify(SubmissionPayload submissionPayload, ConstraintValidatorCo
...
+   * @param submissionPayload the submission paylaod to validate.
+   * @param context           the validator context for keeping track of the constrain violations.
+   * @return whether the protected checkins of this submission payload contain all the same location id.
+   */
+  public boolean verifyHaveSameLocationIdHash(SubmissionPayload submissionPayload, ConstraintValidatorContext context) {
+    final boolean checkInsHaveSameLocationId =
+        submissionPayload.getCheckInProtectedReportsList().stream().map(CheckInProtectedReport::getLocationIdHash)
+            .distinct().count() == 1;
+    if (!checkInsHaveSameLocationId) {
+      logger.warn(SECURITY, "Checkins must contain items that all share the same location id");
```
Comment (FelixRottler) [inline services/submission/src/main/java/app/coronawarn/server/services/submission/checkins/EventCheckInProtectedReportsValidator.java:53]:
```suggestion
      logger.warn(SECURITY, "Checkins protected reports must contain items that all share the same location id");
```

File: services/submission/src/main/java/app/coronawarn/server/services/submission/checkins/EventCheckInProtectedReportsValidator.java [hunk 3]
```diff
@@ -27,7 +34,26 @@ public boolean verify(SubmissionPayload submissionPayload, ConstraintValidatorCo
...
             && verifyEncryptedCheckInRecordLength(checkInProtectedReport, validatorContext))
-        .allMatch(checkInValidation -> Boolean.valueOf(checkInValidation).equals(Boolean.TRUE));
+        .allMatch(checkInValidation -> checkInValidation.equals(Boolean.TRUE));
+  }
+
+  /**
+   * Given a submission in the context of host warnings (submission on behalf) check if all protected checkins share the
+   * same location id.
+   *
+   * @param submissionPayload the submission paylaod to validate.
```
Comment (FelixRottler) [inline services/submission/src/main/java/app/coronawarn/server/services/submission/checkins/EventCheckInProtectedReportsValidator.java:44]:
```suggestion
   * @param submissionPayload the submission payload to validate.
```

File: services/submission/src/test/resources/application.yaml [hunk 4]
```diff
@@ -76,5 +76,5 @@ feign:
   client:
     config:
       default:
-        connect-timeout: 500
-        read-timeout: 500
+        connect-timeout: 900
```
Comment (roesslerj) [inline services/submission/src/test/resources/application.yaml:79]: Was this changed on purpose? Why?

**Review by roesslerj — CHANGES_REQUESTED** [review]

**Review by roesslerj — APPROVED** [review]

**Pull request merged**